coreneuron/io/nrn_setup.cpp

```diff
@@ -756,6 +756,20 @@ void nrn_cleanup() {
                 ml->_thread = nullptr;
             }
 
+            // Probably causes problems with NMODL, which allocates its instance
```

Comment:
This part seems a bit problematic. Both `instance` and `global_variables` are mechanism-specific structs that might be allocated in unified memory. Even putting non-trivial destructors aside, `free` is not necessarily correct if -- for example -- the MOD file translator decides to allocate them in unified memory.

Maybe better would be for the translated MOD file to provide a destructor function that can be called to clean these up.